docs/glossary.md

```diff
@@ -0,0 +1,183 @@
+
+## Change
+
+A change is a commit as it [evolves over time](#rewrite).
+
+## Change ID
+
+A change ID is a unique identifier for a [change](#change). They are typically
+16 bytes long and are often randomly generated. They are presented as
+hexadecimal numbers at the start of the line in `jj log`, using 12 hexadecimal
+digits by default. However, the hexadecimal digits are taken from the end of the
+alphabet (z-k) instead of the usual 0-9 and a-f.
+
+## Commit
+
+A snapshot of the files in the repository at a given point in time (technically
+a [tree object](#tree)), together with some metadata. The metadata includes the
+author, the date, and pointers to the commit's parents. Through the pointers to
+the parents, the commits form a
+[Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph)
+.
+
+Note that even though commits are stored as snapshots, they are often treated
+as differences between snapshots, namely compared to their parent's snapshot. If
+they have more than one parent, then the difference is computed against the
+result of merging the parents. For example, `jj diff` will show the differences
+introduced by a commit compared to its parent(s), and `jj rebase` will apply
+those changes onto another base commit.
+
+The word "revision" is used as a synonym for "commit".
+
+## Commit ID
+
+A commit ID is a unique identifier for a [commit](#commit). They are 20 bytes
+long when using the Git backend. They are presented in regular hexadecimal
+format at the end of the line in `jj log`, using 12 hexadecimal digits by
+default.
+
+## Co-located repos
+
+When using the Git [backend](#backend) and the backing Git repository's `.git/`
+directory is a sibling of `.jj/`, the we call the repository "co-located".
+
+See [here](git-compatibility.md#co-located-jujutsugit-repos) for details.
+
+## Conflict
+
+TODO
+
+Conflicts can occur in files, but they can also occur in branches, for example.
+
+See [here](conflicts.md) for details.
+
+## Divergent change
+
+A divergent change is a [change](#change) that has more than one visible commit.
+
+## Operation
+
+A snapshot of the [visible commits](#visible-commits) and [branches](#branches)
+at a given point in time (technically a [view object](#view)), together with
+some metadata. The metadata includes the username, hostname, timestamps, and
+pointers to the operation's parents.
+
+## Operation log
+
+The [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) formed by
+[operation](#operation) objects, much in the same way that commits form a DAG,
+which is sometimes called the "commit history".
+
+## Repository
+
+Basically everything under `.jj/`, i.e. the full set of [operations](#operation)
+and [commits](#commit).
+
+## Remote
+
+TODO
+
+## Revision
+
+A synonym for [Commit](#commit).
+
+## Revset
+
+Jujutsu supports a functional language for selecting a set of revisions.
+Expressions in this language are called "revsets". See [here](revsets.md) for
+details.
```

Comment:
Do all revsets resolve to sets of revisions? If so maybe a better way to explain it would be "Revset stands for revision set" and then also add the functional language details. If the functional language itself has a name, add that also.

It may be too late now, but I find it a little bit confusing to use the word "revset" to mean 2 things --- both the expressions in the functional language used to select sets of revisions, but also to mean "a set of revisions" that result from evaluating the expression(s) of this same functional language.

Reply:
The simplest thing that comes to mind is to make "revsets" refer to "revision sets" and "revset expressions" as the expressions of the functional language. If "revset expression" is too long, perhaps "revset query" (and you could name the functional language Revset Query Language).

Reply:
True, we could simply call the expressions "revset expression". I started making that change but I felt that it didn't help, mostly because we almost always mean "revset expression" when we say "revset", so it seems like it would be more useful to find a longer term for the set of revisions selected by a revset expression. Feel free to attempt it and send a PR to update the glossary and revsets.md once this PR has landed.